### PR TITLE
[cudapoa] improve error handling in cudapoa kernels

### PR DESCRIPTION
**Pull request by tijyojwad**
Adding error handling for max edge count error Issue observed in #45

### CODE REVIEW

**Comment by tijyojwad** [comment]
@SamStudio8 can you try this PR out on your dataset and let me know if it fixes your problem?

To test this with `racon-gpu` - 
1. Clone ClaraGenomicsAnalysis in a separate repo and checkout this PR
2. Build `racon-gpu` with the following flag:
`-DCLARAGENOMICSANALYSIS_SRC_PATH=<path_to_CGA_repo_from_1>
3. Run `racon-gpu` as usual

**Comment by SamStudio8** [comment]
@tijyojwad Sure, I'll try this today.

**Comment by SamStudio8** [comment]
@tijyojwad Those `breaks` that you've changed to returns ([here](https://github.com/clara-genomics/ClaraGenomicsAnalysis/pull/46/commits/bf07e31bceb9ff147490223a2d1f804931deb87b#diff-f21024f47885113dd4dd3b5366655ff2R106) and [here](https://github.com/clara-genomics/ClaraGenomicsAnalysis/pull/46/commits/bf07e31bceb9ff147490223a2d1f804931deb87b#diff-f21024f47885113dd4dd3b5366655ff2R162)) now stop us reaching the new [edge error condition](https://github.com/clara-genomics/ClaraGenomicsAnalysis/pull/46/commits/bf07e31bceb9ff147490223a2d1f804931deb87b#diff-f21024f47885113dd4dd3b5366655ff2R237). I guess it is not surprising that having a graph with too many edges will have too many nodes.

**Comment by tijyojwad** [comment]
@SamStudio8 the mechanism we built into `racon` with CUDA support was that if the window failed to get polished on the GPU, then it would fall back to getting polished on the CPU instead. That's why the exit code is `0` since the CPU processing of those windows was successful.

**Comment by tijyojwad** [comment]
@SamStudio8 can you confirm that you're seeing this line in output?
```
polished remaining windows on CPU x.y s
```
This will confirm that failed windows are bring processed on the CPU

**Comment by SamStudio8** [comment]
@tijyojwad Yes! I see that output. eg:
```
[racon::CUDAPolisher::polish] polished remaining windows on CPU 19.288 s
```

File: cudapoa/src/cudapoa_kernels.cu [hunk 1]
```diff
@@ -331,6 +333,12 @@ __global__ void generatePOAKernel(uint8_t* consensus_d,
         }
 
         __syncwarp();
+
+        warp_error = __shfl_sync(0xffffffff, warp_error, 0);
```
Comment (akkamesh) [inline cudapoa/src/cudapoa_kernels.cu]: Use `__shfl_sync(FULL_MASK, warp_error, 0)` for better readability. `FULL_MASK` is defined in `cudapoa_kernels.cuh`. 
Also, make the same change to line 217.

File: cudapoa/src/cudapoa_add_alignment.cu [hunk 2]
```diff
@@ -43,13 +44,14 @@ namespace cudapoa
...
- * @return Number of nodes in graph after update
+ * @return Status code for any errors encountered.
  */
 
 template <bool msa = false>
 __device__
-    uint16_t
-    addAlignmentToGraph(uint8_t* nodes,
+    uint8_t
+    addAlignmentToGraph(uint16_t* new_node_count,
```
Comment (akkamesh) [inline cudapoa/src/cudapoa_add_alignment.cu]: Why not pass `new_node_count` in as reference?

File: cudapoa/src/cudapoa_kernels.cu (hunk 1 above)
Comment (tijyojwad) [reply cudapoa/src/cudapoa_kernels.cu]: Absolutely, done!

File: cudapoa/src/cudapoa_add_alignment.cu (hunk 2 above)
Comment (tijyojwad) [reply cudapoa/src/cudapoa_add_alignment.cu]: Passed in as reference now. More habit that I tend to move towards pointers for values that can change

**Comment by tijyojwad** [comment]
rerun tests

**Pull request merged**